packages/core/src/provider/base.ts

```diff
@@ -396,15 +396,19 @@ export class SmartAccountProvider<
     const { maxFeePerGas, maxPriorityFeePerGas } =
       await this._runMiddlewareStack(uoToSubmit, overrides);
 
+    const _maxPriorityFeePerGas = bigIntMax(
+      BigInt(maxPriorityFeePerGas ?? 0n),
+      bigIntPercent(uoToDrop.maxPriorityFeePerGas, 110n)
+    );
     const _overrides: UserOperationOverrides = {
       maxFeePerGas: bigIntMax(
         BigInt(maxFeePerGas ?? 0n),
-        bigIntPercent(uoToDrop.maxFeePerGas, 110n)
-      ),
-      maxPriorityFeePerGas: bigIntMax(
-        BigInt(maxPriorityFeePerGas ?? 0n),
-        bigIntPercent(uoToDrop.maxPriorityFeePerGas, 110n)
+        bigIntPercent(
+          BigInt(uoToDrop.maxFeePerGas) - _maxPriorityFeePerGas,
+          110n
+        ) + _maxPriorityFeePerGas
```

Comment:
since, `maxFeePerGas = baseFeePerGas + maxPriorityFeePerGas`, yeah this makes sense. this way, we don't double count the 10% bump.

however, shouldn't we just use the current pending block's baseFee?

Reply:
my understanding was the bump should be applied to the base fee, as otherwise, we would be double bumping the maxFeePerGas overall.